pkg/pod/v1alpha1/pod.go

```diff
@@ -0,0 +1,111 @@
+package v1alpha1
+
+import v1 "k8s.io/api/core/v1"
+
+// pod holds the api's pod objects
+type pod struct {
+	object *v1.Pod
+}
+
+// PodList holds the list of pod instances
+type PodList struct {
+	Items []*pod
+}
+
+// listBuilder enables building an instance of
+// podlist
+type listBuilder struct {
+	list    *PodList
+	filters predicateList
+}
+
+// WithAPIList builds the list of pod
+// instances based on the provided
+// pod list api instance
+func (b *listBuilder) WithAPIList(pods *v1.PodList) *listBuilder {
+	if pods == nil {
+		return b
+	}
+	for _, p := range pods.Items {
+		b.list.Items = append(b.list.Items, &pod{&p})
+	}
+	return b
+}
+
+// WithAPIList builds the list of pod
+// instances based on the provided
+// pod api instances
+func (b *listBuilder) WithAPIObject(pods ...v1.Pod) *listBuilder {
```

Comment:
WithAPIObject can make use of WithAPIList or vice-versa.
Do not repeat the business logic.

Reply:
done